tests/ui/sized-hierarchy/pointee-supertrait.rs

```diff
@@ -0,0 +1,24 @@
+//@ check-pass
+#![feature(sized_hierarchy)]
+
+use std::marker::PhantomData;
+
+pub trait Bar<'a> {
+    type Foo;
+}
+
+pub struct Foo<'a, T: Bar<'a>> {
+    phantom: PhantomData<&'a T>,
+}
+
+impl<'a, 'b, T> PartialEq<Foo<'b, T>> for Foo<'a, T>
+    where
+        T: for<'c> Bar<'c>,
+        <T as Bar<'a>>::Foo: PartialEq<<T as Bar<'b>>::Foo>,
+{
+    fn eq(&self, _: &Foo<'b, T>) -> bool {
+        loop {}
+    }
+}
+
+fn main() { }
```

Comment:
Hmmm. Where does this test come from?

Reply:
This is reduction of the error that requires the `cfg(bootstrap)` on `PartialEq` - I forget what the exact error I was getting was, but it happened w/ my changes and w/out my changes when adding a blanket `PointeeSized` bound. This motivated the change making `PointeeSized` effectively syntactic sugar for an absence of any bounds whatsoever. I've added a comment describing this.